src/logger/assertions.hpp

```diff
@@ -382,5 +385,25 @@ extern void __print_back_trace();
 #undef ERROR      // may conflict with ERROR macro on windows
 #endif
 
+#define BOOST_ENABLE_ASSERT_HANDLER
+
+namespace boost {
+  inline void assertion_failed(
```

Comment:
Is this not defined in the version of Boost we're using? Or are we relying on this header being included before Boost headers so that we take this version of the code?

Reply:
Ah, that makes sense. Let's move the flag to a global compile flag, in case there are ordering issues in headers being included.

Reply:
Turns out this causes other issues.  I misunderstood; it's actually fine to have this here.